# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### My Lane: Refresh / Content Opportunity Scoring

I have chosen the **Refresh / Content Opportunity Scoring** lane because it focuses on identifying content that may benefit from review or updates based on search performance signals. The goal is to build a data-driven approach for ranking pages according to their potential need for refresh, helping content teams prioritize limited resources more effectively. This lane combines exploratory data analysis, feature engineering, and machine learning to support practical decision-making rather than simply predicting a value. I selected this lane because it aligns with my interest in applying machine learning to solve real-world business problems using search intelligence data.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*



The goal of this project is to help prioritize which content pages should be reviewed first for refresh or improvement based on their search performance. The primary decision is determining which pages deserve immediate attention instead of reviewing every page manually. The users of this output are content managers, SEO analysts, or digital marketing teams who can use the ranked recommendations to plan their work efficiently. If the model recommends the wrong pages, the team may spend time updating low-impact content while overlooking pages with greater improvement potential. Therefore, the project is designed as a decision-support tool that helps prioritize review efforts rather than guaranteeing that updating a page will improve its search performance.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
from pathlib import Path
import pandas as pd

# Find the project root (folder containing 'data')
project_root = Path.cwd()

while project_root != project_root.parent:
    if (project_root / "data").exists():
        break
    project_root = project_root.parent

csv_path = project_root / "data" / "raw" / "content_refresh_anonymized.csv"

print("Loading:", csv_path)

df = pd.read_csv(csv_path)

print("Rows:", len(df))
print("Columns:", len(df.columns))
df.head()

Loading: c:\Users\vinay kumar\OneDrive\Documents\vinay\Vinay-flyrank-ml-internship\data\raw\content_refresh_anonymized.csv
Rows: 30000
Columns: 44


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [4]:
# Basic information
print("Rows:", len(df))
print("Columns:", len(df.columns))
print("\nColumns:")
print(df.columns.tolist())

Rows: 30000
Columns: 44

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [5]:
# Basic statistics

print("Total pages:", df["content_id"].nunique())
print("Average CTR:", df["ctr"].mean())
print("Average impressions (90d):", df["impressions_90d"].mean())

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

Total pages: 30000
Average CTR: 0.5107333333333334
Average impressions (90d): 5200.3663

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30000 unique content pages. The average CTR is 51.07%, and the average 90-day impressions are 5200.3663 . These statistics indicate meaningful variation in search performance, making the dataset suitable for building a content opportunity scoring system.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*



This project analyzes historical search and content performance data to identify patterns that may help prioritize pages for review. The results are intended to support decision-making by ranking pages according to their observed likelihood of benefiting from further investigation. The recommendations are based on historical signals and machine learning, so they should be interpreted as decision-support rather than guaranteed outcomes.

This project cannot prove that refreshing a page will improve its search performance, establish causal relationships, or predict Google's search algorithm. Any findings are observational and directional, and human review is required before taking action on the recommendations.


## Self-check

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all) 
- I encountered a FileNotFoundError while loading the starter dataset because the relative path data/raw/content_refresh_anonymized.csv could not be found from the notebook's current working directory. I need to verify the project structure or use the correct file path before running the analysis.

- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card.